### PR TITLE
Fix duplicate age issue

### PR DESCRIPTION
**Pull request by victoria34**
Fix https://github.com/oncokb/oncokb/issues/1797

Describe changes proposed in this pull request:
- Display age range

image

### CODE REVIEW

File: src/pages/patientView/trialMatch/TrialMatchTableUtils.ts [hunk 1]
```diff
@@ -85,9 +85,41 @@ export function groupTrialMatchesByAgeNumerical(armGroup: ITrialMatch[]): IClini
         clinicalGroupMatch.notMatches = positiveAndNegativeMatches.notMatches;
         return clinicalGroupMatch;
     });
+    if (matches.length > 1) {
+        return removeDuplicationByAge(matches);
+    }
     return matches;
 }
 
+export function removeDuplicationByAge(clinicalGroupMatch: IClinicalGroupMatch[]): IClinicalGroupMatch[] {
```
Comment (alisman) [inline src/pages/patientView/trialMatch/TrialMatchTableUtils.ts]: this function could use a lot more commenting to explain what's going on.

File: src/pages/patientView/trialMatch/TrialMatchTableUtils.ts [hunk 2]
```diff
@@ -85,9 +85,41 @@ export function groupTrialMatchesByAgeNumerical(armGroup: ITrialMatch[]): IClini
...
+    const uniqClinicalGroupMatch: IClinicalGroupMatch[] = _.clone(clinicalGroupMatch);
+    const duplicateIndexes: number[] = [];
+    for (let i = 0; i < clinicalGroupMatch.length - 1; i++) {
+        for (let j = i + 1; j < clinicalGroupMatch.length; j++) {
+            if (!_.isUndefined(clinicalGroupMatch[i]) && !_.isUndefined(clinicalGroupMatch[j]) &&
+                isIdenticalArrays(clinicalGroupMatch[i].trialOncotreePrimaryDiagnosis.positive, clinicalGroupMatch[j].trialOncotreePrimaryDiagnosis.positive) &&
+                isIdenticalArrays(clinicalGroupMatch[i].trialOncotreePrimaryDiagnosis.negative, clinicalGroupMatch[j].trialOncotreePrimaryDiagnosis.negative)) {
+                if (!uniqClinicalGroupMatch[i].trialAgeNumerical.includes(clinicalGroupMatch[j].trialAgeNumerical)) {
+                    uniqClinicalGroupMatch[i].trialAgeNumerical += ',' + clinicalGroupMatch[j].trialAgeNumerical;
```
Comment (alisman) [inline src/pages/patientView/trialMatch/TrialMatchTableUtils.ts]: i think the name of the function should reflect this little operation here.  we aren't just removing duplicates but merging in some sense.

File: src/pages/patientView/trialMatch/TrialMatchTableUtils.ts (hunk 1 above)
Comment (alisman) [reply src/pages/patientView/trialMatch/TrialMatchTableUtils.ts]: i think this code would be clearer and more performant if you first did groupBy using as a comparator the serialized positive and negative array sorted/concated together.  now you have the duplicates together and you can iterate through and do the merge on the age.  then just flatten it.

**Comment by alisman** [comment]
can you squash commits.

File: src/pages/patientView/trialMatch/TrialMatchTableUtils.ts [hunk 3]
```diff
@@ -85,9 +85,27 @@ export function groupTrialMatchesByAgeNumerical(armGroup: ITrialMatch[]): IClini
         clinicalGroupMatch.notMatches = positiveAndNegativeMatches.notMatches;
         return clinicalGroupMatch;
     });
+    if (matches.length > 1) {
+        return mergeClinicalGroupMatchByAge(matches);
+    }
     return matches;
 }
 
+// Merge clinical matches by age when they have the same trialOncotreePrimaryDiagnosis but different trialAgeNumericals.
```
Comment (alisman) [inline src/pages/patientView/trialMatch/TrialMatchTableUtils.ts]: @JiaoJiao123 does this one have a test?

Comment (victoria34) [reply src/pages/patientView/trialMatch/TrialMatchTableUtils.ts]: Not yet. I'll create a test for the function. Btw, I'm @victoria34. You know, I don't want to miss any messages. Thanks

Comment (victoria34) [reply src/pages/patientView/trialMatch/TrialMatchTableUtils.ts]: Added test for mergeClinicalGroupMatchByAge()

**Pull request merged**